flags/flags.go

```diff
@@ -477,9 +477,34 @@ func parseCaURL(ctx *cli.Context, caURL string) (string, error) {
 	if u.Scheme != "https" {
 		return "", errs.InvalidFlagValueMsg(ctx, "ca-url", caURL, "must have https scheme")
 	}
+
+	hostname := u.Hostname()
+	host := hostname
+	if u.Port() != "" {
+		host += ":" + u.Port()
+	}
+	_, _, err = net.SplitHostPort(host)
 	// if host represents a valid IPv6 address, ensure that it contains brackets
-	if ip := net.ParseIP(u.Hostname()); ip != nil && ip.To4() == nil {
-		u.Host = "[" + ip.String() + "]" + ":" + u.Port()
+	if err != nil && strings.Contains(err.Error(), "too many colons in address") {
+		// this is most probably an IPv6 without brackets, e.g. ::1, 2001:0db8:85a3:0000:0000:8a2e:0370:7334
+		// in case a port was appended to this wrong format, we try to extract the port, then check if it's
+		// still a valid IPv6: 2001:0db8:85a3:0000:0000:8a2e:0370:7334:8443 (8443 is the port). If none of
+		// these cases, then the input dns is not changed.
+		lastIndex := strings.LastIndex(host, ":")
+		hostPart, portPart := host[:lastIndex], host[lastIndex+1:]
+		if ip := net.ParseIP(hostPart); ip != nil {
+			hostname = "[" + hostPart + "]:" + portPart
+		} else if ip := net.ParseIP(host); ip != nil {
+			hostname = "[" + host + "]"
+		}
+		u.Host = hostname
+	} else if ip := net.ParseIP(hostname); ip != nil && ip.To4() == nil {
```

Comment:
Can you give an example of a IPv6 address that would not fall into the first case, but would fall into the second case. Just trying to understand how we'd fall into the second case.

Reply:
Is the second case only for IPv6 addresses that were bracketed? I guess, would you mind adding a comment on the second case to just make it abundantly clear what we're checking and why. 

Currently I'm thinking we do that because if the original address had brackets, url.Parse() will remove them. So we want to add them back.

Reply:
I must've overlooked this case when checking the test coverage, because it doesn't seem to be covered by the test cases and I can't currently think of a case that would end up here. They would already fail at parsing the URL. I think I must've changed the first case to cover all weird IPv6 addresses; at least the ones that could already be parsed as an URL.